### PR TITLE
[#2288] Add integration tests verifying forwarding of QoS level in downstream messages

### PR DESCRIPTION
**Pull request by fkaltner**
The HTTP adapter suffered from a bug when a client sent an event and set a device level QoS to something different than 1 or set no device level QoS at all. In any case the event was sent with QoS 1 but the message property indicating the device level QoS to the northbound application (which should always be 1 as this is the only supported QoS for events) was wrong.

Signed-off-by: Florian Kaltner <florian.kaltner@bosch.io>

### CODE REVIEW

File: tests/src/test/java/org/eclipse/hono/tests/http/EventHttpIT.java [hunk 1]
```diff
@@ -52,4 +63,37 @@ protected void assertAdditionalMessageProperties(final Message msg) {
...
+        if (setup.failed()) {
+            ctx.failNow(setup.causeOfFailure());
+            return;
+        }
+
+        httpClient.create(
+                getEndpointUri(),
+                Buffer.buffer("hello"),
+                requestHeaders,
+                ResponsePredicate.status(HttpURLConnection.HTTP_BAD_REQUEST))
```
Comment (fkaltner) [inline tests/src/test/java/org/eclipse/hono/tests/http/EventHttpIT.java:96]: Note that there was a code change which leads to a different behaviour in Hono's AMQP adapter in 1.5.:
Previously the message was accepted (and the QoS was ignored), now it is rejected.

It's not specified as far as I can see how the adapter will react on this, but this may impact client if they had accidentally set a _qos_level_ HTTP header for events to AT_MOST_ONCE.

Comment (sophokles73) [reply tests/src/test/java/org/eclipse/hono/tests/http/EventHttpIT.java:96]: you mean HTTP adapter, right?

File: tests/src/test/java/org/eclipse/hono/tests/IntegrationTestSupport.java [hunk 2]
```diff
@@ -73,7 +73,7 @@
     /**
      * The default number of milliseconds to wait for a response to an AMQP 1.0 performative.
      */
-    public static final int DEFAULT_AMQP_TIMEOUT = 400;
+    public static final int DEFAULT_AMQP_TIMEOUT = 1000;
```
Comment (sophokles73) [inline tests/src/test/java/org/eclipse/hono/tests/IntegrationTestSupport.java:76]: please revert this change, we are already applying a factor > 1 to this value when running on test environment

File: tests/src/test/java/org/eclipse/hono/tests/http/EventHttpIT.java (hunk 1 above)
Comment (fkaltner) [reply tests/src/test/java/org/eclipse/hono/tests/http/EventHttpIT.java:96]: Yes, of course 😅 Sorry for the confusion.
